#### 1.Когда используется критерий Стьюдента, а когда Z –критерий?<br>
Параметрические критерии – это Z, t ( критерий Стьюдента) для сравнения двух выборок. Они предполагают, что выборка взята из нормально распределенной генеральной совокупности, более того предполагают равенство дисперсий в группах.<br>
- Z-критерий - известна сигма (или дисперсия) генеральной совокупности
- t-критерий (критерий Стьюдента) - если сигма генеральной совокупности неизвестна


#### 2.Проведите тест гипотезы. Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 кв. мм.

In [1]:
import numpy as np
from scipy import stats
m=17
a=0.05
n=100
x=17.5
d=4
z=(x-m)/np.sqrt(d/n)
print("z =",z)

z = 2.5


1. H0 -  шарики имеют средний диаметр 17 мм, H1 - шарики имеют средний диаметр 17.5 мм;
2. дисперсия известна - используем z-критерий;
3. z = 2.5, т.е 17.5 лежит от 17.0 в 2.5 сигмах (стандартных ошибках среднего(SE)) вправо;
4. z-табличное для 0.95 лежит в 1.645 сигмах вправо;
5. т.е. z-расчетное > z-табличное, и лежит в области альтернативной гипотизы. При а=0.05 отвергаем нулевую гипотезу H0.

#### 3.Проведите тест гипотезы. Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет: 202, 203, 199, 197, 195, 201, 200, 204, 194, 190. Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%? (Провести двусторонний тест.)

In [2]:
cookies = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
m=200
n=10
a=(100-99)/100/2
a

0.005

In [3]:
x = cookies.mean()
x

198.5

In [4]:
std_c = cookies.std(ddof=1)
std_c

4.453463071962462

In [5]:
tn = (x-m)/(std_c/np.sqrt(n))
tn

-1.0651074037450896

1. H0 -  средний вес пачки печенья составляет 200 г., H1 - средний вес пачки печенья составляет НЕ 200 г.;
2. дисперсия генеральной выборки неизвестна - используем t-критерий;
3. t-расчетное = -1.065, т.е x=198.5 лежит от m=200 в -1.065 сигмах (стандартных ошибках среднего(SE)) влево;
4. t-табличное для a=0.005 (или же, что равноценно для двухстороннего теста а=0.995) для 9 степеней свободы лежит в 3.250 сигмах влево и вправо;
5. т.е. t-расчетное < t-табличное, и лежит в области нулевой гипотизы. При а=0.005 принимаем нулевую гипотезу H0.

#### 4.Есть ли статистически значимые различия в росте дочерей?

Рост матерей 172, 177, 158, 170, 178,175, 164, 160, 169, 165

Рост взрослых дочерей: 173, 175, 162, 174, 175, 168, 155, 170, 160

In [6]:
mothers=np.array([172, 177, 158, 170, 178, 175, 164, 160, 169, 165])
daughters=np.array([173, 175, 162, 174, 175, 168, 155, 170, 160])
m=mothers.mean()
m

168.8

In [7]:
n=daughters.mean()
n

168.0

In [8]:
stats.ttest_ind(mothers,daughters)

Ttest_indResult(statistic=0.24400587215969016, pvalue=0.8101467126543231)

так как длины массивов дочек и матерей не совпадают - функция stats.ttest_rel(mothers,daughters) не работает. Добавим в дочки среднее значение по массиву и посчитаем:

In [9]:
mothers=np.array([172, 177, 158, 170, 178, 175, 164, 160, 169, 165])
daughters=np.array([173, 175, 162, 174, 175, 168, 155, 170, 160, 168])

In [10]:
stats.ttest_rel(mothers,daughters)

Ttest_relResult(statistic=0.40022240757904204, pvalue=0.6983260526026128)

pvalue=0.70 > a=0.05 - статистически значимых различий в росте дочерей нет.